Documentation/README.md

```diff
@@ -57,3 +57,130 @@ tar -xf <tarball-destination> -C <path-to-extraction-root>
 cd <path-to-extraction-root>
 ./build.sh
 ```
+## Building a custom .NET Runtime
+
+You can also build a .NET Core Runtime with custom changes in order to test locally the produced runtime with the affected changes.
+
+The .NET Core Runtime consists of 4 repos, core-setup at the top of the stack (which produces the runtime), standard, corefx (BCL repo), coreclr (Runtime repo). 
+
+This means that in order to get a runtime you'll need to build the source-build repo with the `/p:RootRepo=core-setup` parameter, this will automatically build coreclr, standard and corefx. Source-Build is composed by git submodules, so the first thing you need to run before doing anything is:
+
+```console
+git submodule --init --recursive
+```
+
+This will initialize the submodules and clone the repos under the src folder if they haven't been initialized. Note that this will point the submodule to the branch and commit that the submodule is pointing to. However you can go to each repo indivually, add custom remotes and checkout certain branches containing your changes.
+
+Let's say that you want to produce a runtime that has a new API that it's implementation lives in coreclr and you want to create a .NET Core console app that uses it to test it without waiting for a new SDK or Runtime to be produced by official builds. Imagine we will add an API that just throws in Hashtable. 
+
+Hashtable implementation lives in coreclr and it's APIs are exposed in corefx in the System.Runtime.Extensions.cs contract. So in my local coreclr repo I will just add Hashtable.cs the following API:
+
+```cs
+public static void PNSE()
+{
+    throw new PlatformNotSupportedException();
+}
+```
+
+Then in my local corefx repo let's expose it in System.Runtime.Extensions.cs in the Hashtable class:
+
+```cs
+public static void PNSE() { }
+```
+
+Of course we need to commit this changes in our current repos. Once we've commited this locally, let's move to the source-build repo.
+
+Once in the source-build in our commmand prompt or terminal, lets navigate to `src/coreclr`. Once there, we will need to add a remote to the local coreclr repo, in example:
+```console
+git remote add local C:\repos\coreclr
+```
+
+Then, let's fetch the changes from our repo, by running:
+```console
+git fetch local
+```
+
+Once we've fetch the changes, we just need to checkout the branch where we made the changes:
+```console
+git checkout branchWithChanges
+```
+
+Just repeat this steps under `src/corefx` as well, to pull the corefx changes. Once we've done that, let's navigate to the `source-build` root and run:
+
+```console
+build.{cmd|sh} /p:RootRepo=core-setup
+```
+
+Note that since you've changed the coreclr and corefx commits, the incremental build will find the current commit changed so if you don't want to see messages like:
```

Comment:
> incremental build

Should this just be `build`? It doesn't work incrementally.

Reply:
Makes sense.